internal/controller/warehouses/images.go

```diff
@@ -129,11 +129,20 @@ func imageDiscoveryLogFields(sub kargoapi.ImageSubscription) []any {
 		"platformConstrained", sub.Platform != "",
 	}
 	switch sub.ImageSelectionStrategy {
-	case kargoapi.ImageSelectionStrategySemVer, kargoapi.ImageSelectionStrategyDigest:
-		f = append(
-			f,
-			"semverConstraint", sub.SemverConstraint,
-		)
+	case kargoapi.ImageSelectionStrategySemVer:
+		if sub.Constraint != "" {
+			f = append(
+				f,
+				"constraint", sub.Constraint)
```

Comment:
idk why the original code split this command across multiple lines, but let's not blindly maintain that. Our goal should be maximum readability. In this case, that means this:

```suggestion
			f = append(f, "constraint", sub.Constraint)
```